src/python/gem5/simulate/simulator.py

```diff
@@ -649,3 +728,32 @@ def save_checkpoint(self, checkpoint_dir: Path) -> None:
                                will be saved.
         """
         m5.checkpoint(str(checkpoint_dir))
+
+    def _get_simobject_tree_hash(self) -> int:
+        if self._instantiated:
+            with StringIO() as f:
+                self._root.print_ini(ini_file=f)
+                return hash(f.read())
+
+        return self.hash(self.board)
+
+    def _get_simulator_hash(self) -> int:
+        return sum(
+            hash(attr)
+            for attr in [
+                self._full_system,
+                self._on_exit_event,
+                self._expected_execution_order,
+                self._checkpoint_path,
+            ]
+        )
+
+    def hash(self) -> int:
```

Comment:
This should probably be `__hash__` if you want it to work with dictionaries, sets, and the builtin `hash` function.